test-data/unit/check-optional.test

```diff
@@ -385,7 +385,7 @@ def lookup_field(name, obj):
         attr = None
 
 [case testTernaryWithNone]
-reveal_type(None if bool() else 0)  # E: Revealed type is 'Union[builtins.int, builtins.None]'
+reveal_type(None if bool() else 0)  # E: Revealed type is 'Union[builtins.None, builtins.int]'
```

Comment:
Could you please add few dedicated tests? (Probably not in this test but in `check-unions`)